File: src/main/java/recipestore/input/RecipeApi.java
package recipestore.input;


import org.apache.jena.rdf.model.Resource;
import org.jooq.lambda.Seq;
import recipestore.db.triplestore.TripleStoreDAO;

import javax.inject.Inject;
import java.io.InputStream;
import java.util.stream.Stream;

public class RecipeApi {

    private final TripleStoreDAO tripleStoreDAO;
    private final InputStream datasetStream;

    @Inject
    public RecipeApi(TripleStoreDAO tripleStoreDAO, InputStream datasetStream) {
        this.tripleStoreDAO = tripleStoreDAO;
        this.datasetStream = datasetStream;
    }


    public void loadRecipe(boolean forceCreate) {
        if (forceCreate)
            tripleStoreDAO.delete(true);
        tripleStoreDAO.populate(datasetStream);
    }

    public Stream<Resource> getRecipeData() {
        return Seq.seq(tripleStoreDAO.getResource("http://schema.org/Recipe"));
    }

}

File: src/main/scala/recipestore/input/RecipeApi.scala
package recipestore.input

import java.io.InputStream
import javax.inject.Inject

import org.apache.jena.rdf.model.Resource
import recipestore.db.triplestore.TripleStoreDAO

import scala.collection.JavaConverters._

class RecipeApi @Inject()(val tripleStoreDAO: TripleStoreDAO, val datasetStream: InputStream) {
  def loadRecipe(forceCreate: Boolean) {
    if (forceCreate) tripleStoreDAO.delete(true)
    tripleStoreDAO.populate(datasetStream)
  }

  def getRecipeData: Iterator[Resource] = {
    return tripleStoreDAO.getResource("http://schema.org/Recipe").asScala
  }

  def getRecipeData(resourceFilter: (Resource) => Boolean): Seq[Resource] = {
    return getRecipeData.toSeq.filter(resourceFilter(_))
  }
}